Source/Analytics/Read/Map/MapProcessor.cs

```diff
@@ -0,0 +1,152 @@
+                                                                                                        }  
+                                                                                                    },
+                                                                            CaseReportsLast30Days = new List<CaseReportForMap>{
+                                                                                                        new CaseReportForMap(){
+                                                                                                            CaseReportsId = Guid.NewGuid(),
+                                                                                                            NumberOfPeople = totalCases,
+                                                                                                            Location = LocationForCaseReport
+                                                                                                        }
+                                                                                                    }
+                    };
+                    _caseReportRepository.Insert(casesBeforeDay);
+                }
+                else{
+                    if(casesBeforeDay.CasesPerHealthRisk.ContainsKey(healthRiskId)){
+                        casesBeforeDay.CasesPerHealthRisk[healthRiskId].CaseReportsLast7Days.Add(
+                                                                                            new CaseReportForMap(){ 
+                                                                                                CaseReportsId = Guid.NewGuid(),
+                                                                                                 NumberOfPeople = totalCases, 
+                                                                                                 Location = LocationForCaseReport 
+                                                                                            });
+                        casesBeforeDay.CasesPerHealthRisk[healthRiskId].CaseReportsLast30Days.Add(
+                                                                                            new CaseReportForMap(){ 
+                                                                                                CaseReportsId = Guid.NewGuid(), 
+                                                                                                NumberOfPeople = totalCases, 
+                                                                                                Location = LocationForCaseReport
+                                                                                            });
+                    }else{
+                        casesBeforeDay.CasesPerHealthRisk[healthRiskId]  =  new CaseReportsRetrieved(){  
```

Comment:
And we could also have a constructor for CaseReportRetrieved. Then I think this section of code will look more understandable